test/core/units/malformed_unit_dropdowns.dyn

```diff
@@ -32,7 +32,7 @@
     },
     {
       "ConcreteType": "UnitsUI.Units, UnitsUI",
-      "SelectedIndex": 218,
```

Comment:
Looks like updating to the new forgeUnits will break some existing units nodes.
Looks like we are serializing the index in dropdown nodes which will break if the underlying list of items changes (which happened for this upgrade)
Do you guys think this is serious enough to try to migrate in some way ?
FYI @mjkkirschner @sm6srw

Reply:
Maybe this only happens when the dropDown's SelectedString property does not find a match in the existing itemList.
So an edge case...

Reply:
Ugh...it is weird. Looks like DropDown nodes serialize both the SelectedString and the SelectedIndex. No sure about the order of deserialization. When one is deserialized it tries to override the other....
https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/CoreNodeModels/DropDown.cs#L91-L147

Reply:
@QilongTang Looks like you implemented the serialization of DropDown SelectedString and SelectedIndex.
https://github.com/DynamoDS/Dynamo/pull/9367
Did you expect that the SelectedIndex might become out of sync with SelectedString ?

Reply:
SelectedString is deserialized after SelectedIndex, so when index is out of sync when item sources updated, changed, we fall back to SelectedString later as a second chance. Let me know if that is clear

Reply:
Ok..so this is only an issue when the SelectedString is no longer found. That means it was bad from the start .... or it is missing from the underlying item list, which would be a breaking change (so it might acceptable for it not to work properly)